client/state/data-layer/wpcom/account-recovery/request-reset/index.js

```diff
@@ -1,34 +1,51 @@
 /**
  * Internal dependencies
  */
-import wpcom from 'lib/wp';
-import { ACCOUNT_RECOVERY_RESET_REQUEST } from 'state/action-types';
 import {
-	requestResetSuccess,
-	requestResetError,
-	setResetMethod,
-} from 'state/account-recovery/reset/actions';
+	ACCOUNT_RECOVERY_RESET_REQUEST,
+	ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+	ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
+} from 'state/action-types';
+import { setResetMethod } from 'state/account-recovery/reset/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
 
-export const handleRequestReset = ( { dispatch }, action ) => {
+export const requestReset = ( { dispatch }, action ) => {
 	const {
 		userData,
 		method,
 	} = action;
 
-	wpcom.req.post( {
+	dispatch( http( {
+		method: 'POST',
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/request-reset',
 		body: {
 			...userData,
 			method,
 		},
-		apiNamespace: 'wpcom/v2',
-		path: '/account-recovery/request-reset',
-	} ).then( () => {
-		dispatch( requestResetSuccess() );
-		dispatch( setResetMethod( method ) );
-	} )
-	.catch( ( error ) => dispatch( requestResetError( error ) ) );
+	}, action ) );
+};
+
+export const handleError = ( { dispatch }, action, rawError ) => {
+	dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
+		error: rawError.message,
+	} );
+};
+
+export const handleSuccess = ( { dispatch }, action ) => {
+	const { method } = action;
+	dispatch( setResetMethod( method ) );
+	dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+	} );
 };
 
 export default {
-	[ ACCOUNT_RECOVERY_RESET_REQUEST ]: [ handleRequestReset ],
+	[ ACCOUNT_RECOVERY_RESET_REQUEST ]: [ dispatchRequest(
+		requestReset,
+		handleSuccess,
+		handleError
```

Comment:
As a general naming note, I prefer handlers to be descriptive of what the handlers do, vs using names like `handleSuccess` or `handleError`. For example the success handler might instead be named `setAccountRecoveryMethod`.

No need to update here, but wanted to point this out for next time.